packages/api/amplify_api/lib/src/graphql/web_socket/blocs/web_socket_bloc.dart

```diff
@@ -38,6 +38,9 @@ class WebSocketBloc with AWSDebuggable, AmplifyLoggerMixin {
     required AmplifyAuthProviderRepository authProviderRepo,
     required WebSocketService wsService,
   }) {
+    // Establish HubEvents Stream
+    Amplify.Hub.addChannel(HubChannel.Api, _hubEventsController.stream);
```

Comment:
I would move all this logic up a layer to `AmplifyAPI`. Reason being that hub events are not really the concern of the bloc, but of the plugin. In Auth, we do the same, so the state machine has no knowledge of hub events; the plugin listens to the state machine and translates its state changes to hub events.